jupyter-tensorflow-full/rockcraft.yaml

```diff
@@ -0,0 +1,189 @@
+      - zip
+
+  kubectl:
+    plugin: nil
+    stage-snaps:
+      - kubectl/1.24/stable
+    organize:
+      kubectl: bin/kubectl
+    stage:
+      - bin/kubectl
+
+  nodejs:
+    plugin: nil
+    stage-packages: [nodejs]
+
+  generate-locale:
+    plugin: nil
+    build-packages:
+      - locales
+    override-build: |
+      echo "en_US.UTF-8 UTF-8" > /etc/locale.gen
+      locale-gen
+      mkdir -p $CRAFT_PART_INSTALL/usr/lib/locale $CRAFT_PART_INSTALL/usr/share/i18n/locales
+      cp /usr/lib/locale/locale-archive $CRAFT_PART_INSTALL/usr/lib/locale/locale-archive
+      cp -r /usr/share/i18n/locales $CRAFT_PART_INSTALL/usr/share/i18n/locales
+
+  conda-jupyter:
+    plugin: nil
+    source: https://github.com/kubeflow/kubeflow
+    source-type: git
+    source-tag: v1.7-branch # upstream branch
+    build-packages:
+      - lsb-release
+      - wget
+    build-environment:
+      - MINIFORGE_ARCH: "x86_64"
+      - MINIFORGE_VERSION: "4.10.1-4"
+      - PIP_VERSION: "21.1.2"
+      - PYTHON_VERSION: "3.8.10"
+      - HOME: "/home/jovyan"
+      - CONDA_DIR: "/opt/conda"
+      - CONDA_BIN: "/opt/conda/bin"
+    override-build: |
+      set -xe
+      mkdir -p "$CONDA_DIR" "$HOME"
+
+      # Download the miniforge installer
+      wget -qO miniforge.sh \
+        "https://github.com/conda-forge/miniforge/releases/download/${MINIFORGE_VERSION}/Miniforge3-${MINIFORGE_VERSION}-Linux-${MINIFORGE_ARCH}.sh"
+
+      # Download the miniforge install shasums
+      wget -qO miniforge.sh.sha256 \
+        "https://github.com/conda-forge/miniforge/releases/download/${MINIFORGE_VERSION}/Miniforge3-${MINIFORGE_VERSION}-Linux-${MINIFORGE_ARCH}.sh.sha256"
+
+      # Verify that the downloaded installer is correct
+      echo "$(cat miniforge.sh.sha256 | awk '{ print $1; }') miniforge.sh" | sha256sum --check
+
+      # Run the miniforge installer
+      /bin/sh miniforge.sh -b -f -p "${CONDA_DIR}"
+
+      # Configure conda
+      "${CONDA_BIN}/conda" config --system --set auto_update_conda false
+      "${CONDA_BIN}/conda" config --system --set show_channel_urls true
+
+      echo "conda ${MINIFORGE_VERSION:0:-2}" >> "${CONDA_DIR}/conda-meta/pinned"
+      echo "python ${PYTHON_VERSION}" >> "${CONDA_DIR}/conda-meta/pinned"
+
+      # Install the correct versions of python, conda, pip
+      "${CONDA_BIN}/conda" install -y -q \
+                          python="${PYTHON_VERSION}" \
+                          conda="${MINIFORGE_VERSION:0:-2}" \
+                          pip="${PIP_VERSION}"
+
+      # Clean up
+      "${CONDA_BIN}/conda" update -y -q --all
+      "${CONDA_BIN}/conda" clean -a -f -y
+
+      # Install the jupyter python requirements
+      cp ${CRAFT_PART_SRC}/components/example-notebook-servers/jupyter/requirements.txt jupyter-requirements.txt
+      "${CONDA_BIN}/pip" install --no-cache-dir -r jupyter-requirements.txt
+
+      # Generate a jupyter lab config
+      "${CONDA_BIN}/jupyter" lab --generate-config --allow-root
```

Comment:
why do you need root ? ( i don't know a lot about conda)

Reply:
Need to verify.

Reply:
Will remove.